bin/node-template/node/src/command.rs

```diff
@@ -69,11 +69,19 @@ pub fn run() -> sc_cli::Result<()> {
 	match &cli.subcommand {
 		Some(subcommand) => {
 			let runner = cli.create_runner(subcommand)?;
-			runner.run_subcommand(subcommand, |config| {
-				let PartialComponents { client, backend, task_manager, import_queue, .. }
-					= new_partial(&config)?;
-				Ok((client, backend, import_queue, task_manager))
-			})
+			runner.run_subcommand(
+				subcommand,
+				|config| {
+					let PartialComponents { client, backend, task_manager, import_queue, .. }
+						= new_partial(&config)?;
+					Ok((client, backend, import_queue, task_manager))
+				},
+				|config| {
+					let (task_manager, client, backend, network_status_sinks)
+						= new_light_base(config)?;
+					Ok((client, backend, network_status_sinks, task_manager))
+				},
+			)
```

Comment:
What if we instead did something like this:

```suggestion
      let runner = cli.create_runner(subcommand)?;
			if cli.role == Role::Light {
				runner.run_subcommand(subcommand, |config| {
					let (task_manager, client, backend, import_queue, network_status_sinks, ..) =
						new_light_base(config)?;
					Ok((client, backend, import_queue, network_status_sinks, task_manager))
				});
			} else {
				runner.run_subcommand(subcommand, |config| {
					let PartialComponents { client, backend, task_manager, import_queue, network_status_sinks, ..}
						= new_partial(&config)?;
					Ok((client, backend, import_queue, network_status_sinks, task_manager))
				});
			}
```


Then we'd get rid of the new generics in `run_subcommand` and allow the other subcommands to run for the light client.

Reply:
Ah no you cant. `cli.role` is the one of runcmd, not the one of your subcommand. I know this thing is really a trap we fixed a similar issue in cumulus recently.

TL;DR: don't use cli..... it's not the command line arguments you would expect

Reply:
Would it be possible to have the `--light` flag no matter which command we're running?

Reply:
you can use `runner.config().role` instead

Reply:
Yeah, but that's set by the subcommand, while we want to be able to run (almost) any subcommand on either a light or full node.

At the moment if I run `cargo run -- --light build-spec` I get:

```
error: Found argument 'build-spec' which wasn't expected, or isn't valid in this context

USAGE:
    substrate --light
```

Reply:
We don't support this kind of arguments and also don't want to support them.

Reply:
I don't really see any way I can make this code cleaner then.